lib/node_modules/@stdlib/array/complex64/test/test.reduce.js

```diff
@@ -0,0 +1,213 @@
+});
+
+tape( 'attached to the prototype of the main export is a `reduce` method', function test( t ) {
+	t.strictEqual( hasOwnProp( Complex64Array.prototype, 'reduce' ), true, 'has property' );
+	t.strictEqual( isFunction( Complex64Array.prototype.reduce ), true, 'has method' );
+	t.end();
+});
+
+tape( 'the method throws an error if invoked with a `this` context which is not a complex number array instance', function test( t ) {
+	var values;
+	var arr;
+	var i;
+
+	arr = new Complex64Array( 5 );
+
+	values = [
+		'5',
+		5,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[],
+		function noop() {}
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.map.reduce( value, caddf );
+		};
+	}
+});
+
+tape( 'the method throws an error if provided a first argument which is not a function', function test( t ) {
+	var values;
+	var arr;
+	var i;
+
+	arr = new Complex64Array( 10 );
+
+	values = [
+		'5',
+		3.14,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[]
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.reduce( value );
+		};
+	}
+});
+
+tape( 'the method throws an error if not provided an initial value when operating on an empty complex number array', function test( t ) {
+	var arr;
+
+	arr = new Complex64Array( 0 );
+	t.throws( foo, TypeError, 'throws an error' );
+	t.end();
+
+	function foo() {
+		arr.reduce( caddf );
+	}
+});
+
+tape( 'the method uses the first element of the array as the initial value when an initial value is not provided', function test( t ) {
+	var valueArray;
+	var accArray;
+	var expected;
+	var actual;
+	var arr;
+	var ind;
+
+	arr = new Complex64Array( [ 1.0, -1.0, 2.0, -2.0, 3.0, -3.0 ] );
+	accArray = [ 1.0, -1.0, 3.0, -3.0 ];
+	valueArray = [ 2.0, -2.0, 3.0, -3.0 ];
+	expected = new Complex64( 6.0, -6.0 );
+	actual = arr.reduce( reducer );
+
+	t.strictEqual( instanceOf( actual, Complex64 ), true, 'returns expected value' );
+	t.strictEqual( realf( actual ), realf( expected ), 'returns expected value' );
+	t.strictEqual( imagf( actual ), imagf( expected ), 'returns expected value' );
+
+	function reducer( acc, value, index ) {
+		ind = 2*(index-1);
+		t.strictEqual( instanceOf( acc, Complex64 ), true, 'provides expected value' );
+		t.strictEqual( realf( acc ), accArray[ ind ], 'provides expected value' );
+		t.strictEqual( imagf( acc ), accArray[ ind+1 ], 'provides expected value' );
+		t.strictEqual( instanceOf( value, Complex64 ), true, 'provides expected value' );
+		t.strictEqual( realf( value ), valueArray[ ind ], 'provides expected value' );
+		t.strictEqual( imagf( value ), valueArray[ ind+1 ], 'provides expected value' );
```

Comment:
```suggestion
		t.strictEqual( instanceOf( acc, Complex64 ), true, 'returns expected value' );
		t.strictEqual( realf( acc ), accArray[ ind ], 'returns expected value' );
		t.strictEqual( imagf( acc ), accArray[ ind+1 ], 'returns expected value' );
		t.strictEqual( instanceOf( value, Complex64 ), true, 'returns expected value' );
		t.strictEqual( realf( value ), valueArray[ ind ], 'returns expected value' );
		t.strictEqual( imagf( value ), valueArray[ ind+1 ], 'returns expected value' );
```

We use "returns expected value" as the default test message.